test/test_coordinates_space.py

```diff
@@ -0,0 +1,325 @@
+            lisa_Y_fd = -numpy.sqrt(6)/3 * bbhx_fd['LISA_E'] + \
+                        numpy.sqrt(3)/3 * bbhx_fd['LISA_T']
+            lisa_Z_fd = numpy.sqrt(2)/2 * bbhx_fd['LISA_A'] + \
+                        numpy.sqrt(6)/6 * bbhx_fd['LISA_E'] + \
+                        numpy.sqrt(3)/3 * bbhx_fd['LISA_T']
+            channel_xyz_fd = {'LISA_X':lisa_X_fd,
+                              'LISA_Y':lisa_Y_fd,
+                              'LISA_Z':lisa_Z_fd}
+
+            t_geo, ra, dec, polarization_geo = \
+                ssb_to_geo(params['tc'], params['eclipticlongitude'], 
+                           params['eclipticlatitude'], params['polarization'])
+
+            params_3g = params.copy()
+            params_3g['tc'] = t_geo
+            params_3g['ra'] = ra
+            params_3g['dec'] = dec
+            params_3g['polarization'] = polarization_geo
+            params_3g['coa_phase'] = numpy.pi/2 - params['coa_phase']
+
+            E3_signal = strain_generator(det='D1', model='IMRPhenomD',
+                                         fs=1./params_3g['delta_t'],
+                                         flow=params_3g['f_lower'],
+                                         fref=params_3g['f_ref'],
+                                         tc=params_3g['tc'],
+                                         params=params_3g,
+                                         fd_taper=False)          
+            E3_signal_fd = {
+                'DIY_E3':E3_signal[0].to_frequencyseries(params_3g['delta_f'])
+            }
+
+            index_E3 = numpy.where(numpy.abs(
+                        E3_signal_fd['DIY_E3'].sample_frequencies -
+                        params_3g['f_ref']) < 1e-7)
+            index_LISA3 = numpy.where(numpy.abs(
+                        bbhx_fd['LISA_A'].sample_frequencies - 
+                        params['f_ref']) < 1e-7)
+            amp_E3 = numpy.abs(E3_signal_fd['DIY_E3'])[index_E3[0][0]]
+            amp_LISA3 = numpy.abs(channel_xyz_fd['LISA_Z'][index_LISA3[0][0]])
+            phase_E3 = numpy.rad2deg(numpy.angle(
+                        E3_signal_fd['DIY_E3'][index_E3[0][0]]))
+            phase_LISA3 = numpy.rad2deg(numpy.angle(
+                        channel_xyz_fd['LISA_Z'][index_LISA3[0][0]]))
+
+            amp_E3_psi.append(amp_E3)
+            phase_E3_psi.append(phase_E3)
+            amp_LISA3_psi.append(amp_LISA3)
+            phase_LISA3_psi.append(phase_LISA3)
+
+        dist_amp = curve_similarity(amp_E3_psi/numpy.mean(amp_E3_psi),
+                                    amp_LISA3_psi/numpy.mean(amp_LISA3_psi))
+        dist_phase = curve_similarity(numpy.array(phase_E3_psi),
+                                      numpy.array(phase_LISA3_psi))
+        if (numpy.abs(dist_amp - 0.2838670151034317) < 1e-2) and \
+            (numpy.abs(dist_phase - 151.77197349820668) < 1e-2):
+            passed = True
+        else:
+            passed = False
```

Comment:
Can you add some inline documentation here to explain this test a bit more? It's a bit opaque to someone  not already familiar with what you are trying to do. 

Separately, where do these random numbers come from, for example, how would one update these if needed? It would be good if this is also clear in the inline docs

Reply:
@ahnitz I have added a paragraph, please check.